File: 5.implementation/NXP/redsun-service/src/main/java/com/redsun/service/service/CronExecutionsService.java
package com.redsun.service.service;

import com.redsun.service.entities.CronExecutions;
import com.redsun.service.entities.Result;

/**
 * CronExecutions Service interface
 */
public interface CronExecutionsService {
	
    // Insert.
	Result insert(final CronExecutions cronExecutions) ;

    // Update.
	Result update(final CronExecutions cronExecutions) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listCronExecutionsForPageAndFilter(final int itemsPerPage, final int pageNo, final CronExecutions cronExecutions);

}

File: 5.implementation/VT/redsun/src/main/webapp/static/js/services/CronExecutionsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('cronExecutionService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(cronExecution) {
			var urlServer = $rootScope.makePostURL("/cronexecution/insert");
			return $http.post(urlServer, cronExecution);
		}
		
		// Update.
		, update: function(cronExecutionId, cronExecution) {
			var urlServer = $rootScope.makePostURL("/cronexecution/update/" + cronExecutionId);
			return $http.put(urlServer, cronExecution);
		}
		
		// Delete.
		, delete: function(cronExecutionId) {
			var urlServer = $rootScope.makePostURL("/cronexecution/delete/" + cronExecutionId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(cronExecutionId) {
			var url = $rootScope.makeGetURL("/cronexecution/getbyid/" + cronExecutionId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, cronExecution) {
			var url = $rootScope.makeGetURL("/cronexecution/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: cronExecution });
		}
		
	}// return.
});